etcdserver/api/v3rpc/metrics.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v3rpc
+
+import (
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+var (
+	receivedCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "etcd",
+			Subsystem: "grpc",
+			Name:      "requests_total",
+			Help:      "Counter of received requests.",
+		}, []string{"grpc_service", "grpc_method"})
+
+	failedCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "etcd",
+			Subsystem: "grpc",
+			Name:      "requests_failed_total",
+			Help:      "Counter of failed requests.",
+		}, []string{"grpc_service", "grpc_method", "grpc_code"})
```

Comment:
How many possible codes are there?

One of the things you need to be careful with with labels is that if there's never a failure here, then the time series will never be exposed. This can cause fun breakage, as there's no timeseries to match on when doing say division to get the failure ratio so you'll just get an empty result.

The usual way to handle this is to ensure all the labels you expect to exist are created in advance. In Go you usually do this by calling just `WithLabelValues` on everything you expect to exist as part of initializing.